### ISSUE TITLE
Any version of Steam Proton past version 5.0-10 causes games to not launch.

### ISSUE THREAD

**Issue by danigtrz** [NixOS/nixpkgs#207374]
Every Windows game I have tried does not launch with any version of Proton past version 5.0-10.

I am using Flatpak Steam and am on the unstable branch of NixOS.

My system: https://github.com/danigtrz/nixos

I used `sudo flatpak override --env=__NV_PRIME_RENDER_OFFLOAD=1 --env=__NV_PRIME_RENDER_OFFLOAD_PROVIDER=NVIDIA-G0 --env=__GLX_VENDOR_LIBRARY_NAME=nvidia --env=__VK_LAYER_NV_optimus=NVIDIA_only com.valvesoftware.Steam` to use my laptop's discrete GPU. The problem still ocurred before using this command.

Here, I run Steam and try to launch game using Proton 7.0.
```
flatpak run com.valvesoftware.Steam 
INFO:root:https://github.com/flathub/com.valvesoftware.Steam/wiki
INFO:root:Will set XDG dirs prefix to /home/rafa/.var/app/com.valvesoftware.Steam
DEBUG:root:Checking input devices permissions
INFO:root:Overriding TZ to America/Tijuana
steam.sh[2]: Running Steam on org.freedesktop.platform 22.08 64-bit
steam.sh[2]: STEAM_RUNTIME is enabled automatically
setup.sh[77]: Steam runtime environment up-to-date!
steam.sh[2]: Steam client's requirements are satisfied
[2022-12-23 04:34:43] Startup - updater built Dec 15 2022 21:26:49
[2022-12-23 04:34:43] Startup - Steam Client launched with: '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam' '-no-cef-sandbox'
Installing breakpad exception handler for appid(steam)/version(1671236931)
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2022-12-23 04:34:43] Loading cached metrics from disk (/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/package/steam_client_metrics.bin)
[2022-12-23 04:34:43] Using the following download hosts for Public, Realm steamglobal
[2022-12-23 04:34:43] 1. https://client-update.akamai.steamstatic.com, /, Realm 'steamglobal', weight was 1000, source = 'update_hosts_cached.vdf'
[2022-12-23 04:34:43] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'update_hosts_cached.vdf'
[2022-12-23 04:34:43] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1671236931)
[2022-12-23 04:34:43] Checking for update on startup
[2022-12-23 04:34:43] Checking for available updates...
[2022-12-23 04:34:43] Downloading manifest: https://client-update.akamai.steamstatic.com/steam_client_ubuntu12
[2022-12-23 04:34:43] Manifest download: send request
Installing breakpad exception handler for appid(steam)/version(1671236931)
[2022-12-23 04:34:43] Manifest download: waiting for download to finish
[2022-12-23 04:34:43] Manifest download: finished
[2022-12-23 04:34:43] Download skipped: /steam_client_ubuntu12 version 1671236931, installed version 1671236931, existing pending version 0
[2022-12-23 04:34:43] Nothing to do
[2022-12-23 04:34:43] Verifying installation...
[2022-12-23 04:34:43] Performing checksum verification of executable files
[2022-12-23 04:34:44] Verification complete
Loaded SDL version 2.27.0-p7692409
XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xf358a8f0
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xf35891c0
ComputeStartupMode: found registry default startup mode: 0Installing breakpad exception handler for appid(steam)/version(1671236931)
steamwebhelper.sh[160]: Runtime for steamwebhelper: defaulting to /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
steamwebhelper.sh[160]: Running under Flatpak, disabling sandbox
steamwebhelper.sh[160]: CEF sandbox already disabled
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
CAppInfoCacheReadFromDiskThread took 111 milliseconds to initialize

(steam:155): Gtk-WARNING **: 04:34:48.261: gtk_disable_setlocale() must be called before gtk_init()
Installing breakpad exception handler for appid(steam)/version(1671236931)
local (potentially out of sync) copy of roaming config loaded - 1556 bytes.
roaming config store loaded successfully - 1556 bytes.
migrating temporary roaming config store
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
BRefreshApplicationsInLibrary 1: 16ms
Installing breakpad exception handler for appid(steam)/version(1671236931)
Failed to init SteamVR because it isn't installed
ExecCommandLine: "'/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam' '-no-cef-sandbox'"
System startup time: 5.75 seconds
Installing breakpad exception handler for appid(steam)/version(1671236931)
BuildCompleteAppOverviewChange: 233 apps
RegisterForAppOverview 1: 3ms
RegisterForAppOverview 2: 3ms
Installing breakpad exception handler for appid(steam)/version(1671236931)
Installing breakpad exception handler for appid(steam)/version(1671236931)
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
Proton: Upgrading prefix from 5.0-1 to 7.0-100 (/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/compatdata/1687950/)
Proton: Removing ShellExecute DDE registry entries.
Proton: Removing stale builtin /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/compatdata/1687950/pfx//drive_c/windows/system32/amd_ags_x64.dll
Proton: Removing stale builtin /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/compatdata/1687950/pfx//drive_c/windows/syswow64/amd_ags_x64.dll
esync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
wine: configuration in L"/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/compatdata/1687950/pfx" has been updated.
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 1687950, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 1687950, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
/bin/sh\0-c\0/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'\0
Game process added : AppID 1687950 "/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'", ProcID 608, IP 0.0.0.0:0
chdir /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 1687950, ActionID 1] : LaunchApp changed task to Completed with ""
pid 612 != 609, skipping destruction (fork without exec?)

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.

(steam:155): LIBDBUSMENU-GLIB-WARNING **: 04:35:27.129: Trying to remove a child that doesn't believe we're it's parent.
esync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
ThreadGetProcessExitCode: no such process 706
                                             ThreadGetProcessExitCode: no such process 704
                                                                                          ThreadGetProcessExitCode: no such process 703
                                   ThreadGetProcessExitCode: no such process 702
                                                                                ThreadGetProcessExitCode: no such process 610
                         ThreadGetProcessExitCode: no such process 709
                                                                      Game process updated : AppID 1687950 "/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'", ProcID 705, IP 0.0.0.0:0
         Setting breakpad minidump AppID = 1687950
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198930137466 [API loaded no]
Game process updated : AppID 1687950 "/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'", ProcID 773, IP 0.0.0.0:0
                                       Setting breakpad minidump AppID = 1687950
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198930137466 [API loaded no]
Game process updated : AppID 1687950 "/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'", ProcID 773, IP 0.0.0.0:0
                                       SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198930137466 [API loaded no]
ThreadGetProcessExitCode: no such process 755
                                             ThreadGetProcessExitCode: no such process 746
                                                                                          ThreadGetProcessExitCode: no such process 740
                                   ThreadGetProcessExitCode: no such process 734
                                                                                ThreadGetProcessExitCode: no such process 714
                         ThreadGetProcessExitCode: no such process 711
                                                                      ThreadGetProcessExitCode: no such process 705
               ThreadGetProcessExitCode: no such process 701
                                                            pid 707 != 706, skipping destruction (fork without exec?)
Game process removed: AppID 1687950 "/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1687950 -- /home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/rafa/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/P5R/P5R.exe'", ProcID 773 
ThreadGetProcessExitCode: no such process 773
ThreadGetProcessExitCode: no such process 722
ThreadGetProcessExitCode: no such process 609
Game 1687950 created interface STEAMAPPS_INTERFACE_VERSION006 / Apps
Game 1687950 created interface SteamController008 / 
Game 1687950 created interface SteamInput002 / 
Game 1687950 created interface SteamUser017 / User
Game 1687950 created interface SteamUser021 / 
Game 1687950 created interface SteamUser021 / User
Game 1687950 created interface SteamUtils010 / 
Game 1687950 method call count for IClientAppManager::GetCurrentLanguage : 1
Game 1687950 method call count for IClientAppManager::GetAppStateInfo : 1
Game 1687950 method call count for IClientUtils::RecordSteamInterfaceCreation : 10
Game 1687950 method call count for IClientUtils::GetAPICallResult : 1
Game 1687950 method call count for IClientUtils::GetAppID : 13
Game 1687950 method call count for IClientUser::GetEncryptedAppTicket : 1
Game 1687950 method call count for IClientUser::RequestEncryptedAppTicket : 1
Game 1687950 method call count for IClientUser::GetSteamID : 6
Game 1687950 method call count for IClientUser::BLoggedOn : 1
Uploaded AppInterfaceStats to Steam
```

**Comment by Shawn8901** [comment]
I have noticed that you are using wayland and it seems for me that you are enforcing wayland renderer in your home variables by setting SDL_VIDEODRIVER to wayland.
For me i need to either set SDL_VIDEODRIVER to x11 for linux natives and SDL_VIDEODRIVER to windib in case proton is using SDL.
You could check if that fixes your game also.

**Comment by m911t** [comment]
this is most likely the case, that env var breaks many games, one of them being PD2

**Comment by danigtrz** [comment]
Unfortunately this did not work.

I removed any variable that mentioned wayland, and tried various combinations of games, proton versions, and both 'x11' and 'windib'. Nothing worked. I also downloaded and tried using Proton GE but that did not work either.

**Comment by jacobranson** [comment]
I have found that doing the following makes my games work under Proton again. For reference, I have tested with Hogwarts Legacy and Red Dead Redemption 2.

- In the Steam UI, search your library for "Steam Linux Runtime 2.0 (soldier)".
- Right click it, click "Properties".
- Click the "Betas" tab, and select "Previous Release" in the dropdown.

This is the same workaround as listed in this closed issue #130699. I wonder if something has regressed?
